ross/defects/misalignment.py

```diff
@@ -0,0 +1,1078 @@
+"""Misalignment module.
+
+This module defines misalignments of various types on the shaft coupling. There are 
+a number of options, for the formulation of 6 DoFs (degrees of freedom).
+"""
+import time
+from abc import ABC, abstractmethod
+
+import numpy as np
+import plotly.graph_objects as go
+import scipy as sp
+import scipy.integrate
+import scipy.linalg
+
+import ross
+from ross.results import TimeResponseResults
+from ross.units import Q_
+
+from .abs_defect import Defect
+from .integrate_solver import Integrator
+
+__all__ = [
+    "MisalignmentFlex",
+    "MisalignmentRigid",
+]
+
+
+class MisalignmentFlex(Defect):
+    """A flexible coupling with misalignment of some kind.
+    
+    Calculates the dynamic reaction force of hexangular flexible coupling 
+    induced by 6DOF's rotor parallel and angular misalignment.
+
+    Parameters
+    ----------
+    dt : float
+        Time step.
+    tI : float
+        Initial time.
+    tF : float
+        Final time.
+    kd : float
+        Radial stiffness of flexible coupling.
+    ks : float
+        Bending stiffness of flexible coupling.
+    eCOUPx : float
+        Parallel misalignment offset between driving rotor and driven rotor along X direction.
+    eCOUPy : float
+        Parallel misalignment offset between driving rotor and driven rotor along Y direction.
+    misalignment_angle : float
+        Angular misalignment angle.
+    TD : float
+        Driving torque.
+    TL : float
+        Driven torque.
+    n1 : float
+        Node where the misalignment is ocurring.
+    speed : float
+        Operational speed of the machine.
+    massunb : array
+        Array with the unbalance magnitude. The unit is kg.m.
+    phaseunb : array
+        Array with the unbalance phase. The unit is rad.
+    mis_type: string
+        String containing the misalignment type choosed. The avaible types are: parallel, by default; angular; combined.
+    
+    Returns
+    -------
+    A force to be applied on the shaft.
+
+    References
+    ----------
+    .. [1] 'Xia, Y., Pang, J., Yang, L., Zhao, Q., & Yang, X. (2019). Study on vibration response 
+    and orbits of misaligned rigid rotors connected by hexangular flexible coupling. Applied 
+    Acoustics, 155, 286-296 ..
+
+    Examples
+    --------
+    >>> from ross.defects.misalignment import misalignment_flex_parallel_example
+    >>> misalignment = misalignment_flex_parallel_example()
+    >>> probe1 = (14, 0)
+    >>> probe2 = (22, 0)
+    >>> response = rotor.run_defect(misalignment)
+    >>> results = misalignment.run_time_response()
```

Comment:
For the API I was thinking of something like:
```python
misalignment_results = rotor.run_misalignment(type='parallel')
```

Reply:
Oh i agree. @bela-bs @hiagopinacio @marcusfilipesr , we should change our standard to this.

Reply:
Hi, @raphaeltimbo, the type of the misalignment (in this case just of the misalignment with flexible coupling) is already included in the input parameters of the class.
ex: misalignment = MisalignmentFlex(dt=0.0001, tI=0, tF=30, ...,  **mis_type="combined"**,)
Are you have some other suggestions for this?

Reply:
Hi @bela-bs ! 
My point here is that the user should not worry about creating the misalignment object, they should only call the a `.run_misalignment()` method and the creation of this misalignment object and calculations should be done internally:

```python
# current code is something like this
from ross.defects.misalignment import misalignment_flex_parallel_example
misalignment = misalignment_flex_parallel_example()
probe1 = (14, 0)
probe2 = (22, 0)
response = rotor.run_defect(misalignment)
results = misalignment.run_time_response()

# my proposal is to have only this:
misalignment_results = rotor.run_misalignment(type='parallel', arg1=..., arg2=...,)

# the run_misalignment method would take care of the intermidiate steps
def run_misalignment(self, type=None, arg1=None, arg2=None, ...):
    misalignment = MisalignmentFlex(dt=0.0001, tI=0, tF=30, ..., mis_type=type,)
    ...
    return results
```

Reply:
Hi @raphaeltimbo, we've just updated the misalignment class call as you have requested. Any additional modifications that you guys think are required, we are totally available.